commitlintplugins.test.ts

```diff
@@ -219,3 +219,23 @@ test('type-space-after-comma2', () => {
     expect(typeSpaceAfterComma2.status).toBe(0);
 });
 
+
+test('type-space-before-paren1', () => {
+    let commitMsgWithNoSpaceBeforeParen = "foo (bar): bla bla bla";
+    let typeSpaceBeforeParen1 = runCommitLintOnMsg(commitMsgWithNoSpaceBeforeParen);
+    expect(typeSpaceBeforeParen1.status).not.toBe(0);
+});
+
+
+test('type-space-before-paren2', () => {
+    let commitMsgWithNoSpaceBeforeParen = "foo(bar): bla bla bla";
+    let typeSpaceBeforeParen2 = runCommitLintOnMsg(commitMsgWithNoSpaceBeforeParen);
+    expect(typeSpaceBeforeParen2.status).toBe(0);
+});
+
+
+test('type-space-before-paren3', () => {
+    let commitMsgWithNoSpaceBeforeParen = "(bar): bla bla bla";
```

Comment:
last thing to do in this PR: as we found this case above as an edge case for this rule, because it's an invalid title (as it has sub-area but no area), but shouldn't be covered by this rule, let's add a TODO item in the list that explains we need a rule for this

Reply:
Should I remove this test? Because as you said, it's not a correct title, but in this test, we expect the status code to be 0.

Reply:
good question, maybe we should remove it (or rather rename it) when we implement the new rule that will take care of titles with sub-area but without area

Reply:
Right